iOS_SDK/OneSignalSDK/Source/OSMessagingController.m

```diff
@@ -372,7 +426,10 @@ - (void)setDataForRedisplay:(OSInAppMessage *)message {
 
             [self.seenInAppMessages removeObject:message.messageId];
             [self.impressionedInAppMessages removeObject:message.messageId];
+            [self.viewedPageIDs removeAllObjects];
```

Comment:
this line shouldn't instead remove all objects from the message page ids?

Reply:
removed tracking from the message itself

Reply:
instead of removing all objects, should only remove the ids that have this IAM id prefix, right?

Reply:
It actually doesn't matter since at this point we have dismissed and are showing a different IAM, at which point we can resend the impressions so either way it will always be empty when a new IAM is showing. I would rather have some weird edge case that results in sending too many impressions than an edge case that results in not sending impressions.